subcommands/targets/offline-update.go

```diff
@@ -0,0 +1,286 @@
+package targets
+
+import (
+	"archive/tar"
+	"compress/bzip2"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"github.com/foundriesio/fioctl/client"
+	"github.com/foundriesio/fioctl/subcommands"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+	"io"
+	"os"
+	"path"
+	"strconv"
+	"strings"
+	"time"
+)
+
+var (
+	factory string
+	prod    bool
+	tag     string
+)
```

Comment:
These are globals for the entire `subcommands.targets` subpackage.
So I'd recommend to either prefix them somehow (to make less generic) or use the `XxxVar` rather than `XxxVarP` below and pass into functions explicitly.
I generally prefer the second way as it is safer (although more verbose and less convenient to someone).

Reply:
I'll do the explicit way.

Reply:
done.